scripts/undemultiplexed_index.py

```diff
@@ -51,165 +41,208 @@
-        target_file.udf['% One Mismatch Reads (Index)'] = float(sample_info['% One Mismatch Reads (Index)'])
-        target_file.udf['% of Raw Clusters Per Lane'] = float(sample_info['% of raw clusters per lane'])
-        target_file.udf['%PF'] = float(sample_info['% PF'])
-        target_file.udf['Ave Q Score'] = float(sample_info['Mean Quality Score (PF)'])
-        Yield_PF_Gb = np.true_divide(float(sample_info['Yield (Mbases)'].replace(',','')), 1000)
-        target_file.udf['Yield PF (Gb)'] = Yield_PF_Gb 
-        target_file.udf['% Perfect Index Read'] = float(sample_info['% Perfect Index Reads'])
-        if not dict(target_file.udf.items()).has_key('% Bases >=Q30'):
-            target_file.udf['% Bases >=Q30'] = float(sample_info['% of >= Q30 Bases (PF)'])
-        if not dict(target_file.udf.items()).has_key('# Reads'):
-            target_file.udf['# Reads'] = float(sample_info['# Reads'].replace(',',''))
-        if self.single:
-            self.read_pairs = int(target_file.udf['# Reads'])
+    def _get_threshold_Q30(self):
+        if self.user_def_tresh.has_key('Threshold for % bases >= Q30'):
+            self.Q30_treshold = self.user_def_tresh['Threshold for % bases >= Q30']
+            qc_logg = ("THRESHOLD FOR %Q30 was set by user to {0}.".format(
+                        self.Q30_treshold))
+            print >> self.qc_log_file, qc_logg
         else:
-            self.read_pairs = np.true_divide(float(target_file.udf['# Reads']),2)
-        target_file.udf['# Read Pairs'] = self.read_pairs
-        target_file.qc_flag = self._QC(target_file, sample_info)
-        set_field(target_file)
-
-    def _QC(self, target_file, sample_info):
-        """Makes per sample warnings if any of the following holds: 
-        % Perfect Index Reads < 60
-        % of >= Q30 Bases (PF) < 80
-        # Reads < 100000
-        OBS: Reads from target file udf if they are already set. Otherwise from 
-        file system!!! This is to take into account yield and quality after 
-        quality filtering if performed.
-        
-        Sets the include reads field"""
-        perf_ind_read = float(sample_info['% Perfect Index Reads'])
-        Q30 = float(sample_info['% of >= Q30 Bases (PF)'])
-        self._get_QC_thresholds()
-        QC1 = (perf_ind_read >= self.QC_thresholds['perf_ind'])
-        QC2 = (Q30 >= self.QC_thresholds['%Q30'])
-        QC3 = (self.read_pairs >= self.QC_thresholds['nr_read'])
-        if QC1 and QC2 and QC3:
-            target_file.udf['Include reads'] = 'YES'
-            return 'PASSED'
-        else:
-            target_file.udf['Include reads'] = 'NO'
-            return 'FAILED'
+            warning = ("Un recognized read length: {0}. Report this to "
+                "developers! set Threshold for % bases >= Q30 if you want to "
+                "run bcl conversion and demultiplexing anyway.".format(self.read_length))
+            if self.run_type == 'MiSeq':
```

Comment:
@Galithil I get your point. I still do not like the idea to hard-code threshold in scripts but you are right in saying that this will add complexity that right now is not beneficial